eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -258,8 +257,9 @@ object Validation {
     }
   }
 
-  def handleChannelUpdate(d: Data, db: NetworkDb, routerConf: RouterConf, origins: Set[GossipOrigin], u: ChannelUpdate, wasStashed: Boolean = false)(implicit ctx: ActorContext, log: LoggingAdapter): Data = {
+  def handleChannelUpdate(d: Data, db: NetworkDb, routerConf: RouterConf, origins: Set[GossipOrigin], update: Either[LocalChannelUpdate, ChannelUpdate], wasStashed: Boolean = false)(implicit ctx: ActorContext, log: LoggingAdapter): Data = {
```

Comment:
This change create potential inconsistency with the `origins` field unfortunately.

`update == Left(_)` <=> `origins == Set(LocalGossip)`

And also confusion between `handleChannelUpdate` vs `handleLocalChannelUpdate`.

Reply:
Yes for a `LocalChannelUpdate` you need to set origins to `Local`, but that doesn't feel too dangerous, the risk of an inconsistency appearing here is quite low.

Do you think we should merge `handleChannelUpdate` and `handleLocalChannelUpdate` in a single function (at the risk of having a huge function)?

Reply:
Actually it's quite a mess to group everything in a single method, because `handleLocalChannelUpdate` may apply some changes to the `Data` before calling `handleChannelUpdate`...

Reply:
Changing the signature to:
```suggestion
  def handleChannelUpdate(d: Data, db: NetworkDb, routerConf: RouterConf, update: Either[LocalChannelUpdate, (Set[RemoteGossip], ChannelUpdate)], wasStashed: Boolean = false)(implicit ctx: ActorContext, log: LoggingAdapter): Data = {
```
would solve the inconsistency, not sure if it is better overall though.

Reply:
> Do you think we should merge handleChannelUpdate and handleLocalChannelUpdate in a single function (at the risk of having a huge function)?

No I don't think so

Reply:
Let me know what you think of https://github.com/ACINQ/eclair/pull/1382/commits/cd37fa1ab42e04d78ca183931ee6c1ed818d10be

Reply:
We can't directly use a `Set[RemoteGossip]`, we have to keep a `Set[GossipOrigin]` for `channel_update`s we receive via payment failures (they are remote, but we don't have an origin for them). Maybe in that case we could use an empty set instead, but that's something I'd like to keep for another PR as it's getting away from the core of this PR.